presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -266,6 +263,25 @@ default void finishDelete(ConnectorSession session, ConnectorTableHandle tableHa
     }
 
     /**
+     * Notifies when select query starts
+     *
+     * @return serializable {@link ConnectorTableHandle}. This handle will be brought back in {@link #finishSelect(ConnectorSession, ConnectorTableHandle)} if null returned, no finishSelect will be called.
+     */
+    default ConnectorTableHandle beginSelect(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> layoutHandle, Collection<ColumnHandle> columnHandles)
```

Comment:
I don't understand why the connection would not be per transaction. It would have to be in order to respect the isolation level. Also, a single Presto query can access multiple tables (or the same table multiple times), so I don't understand the distinction between Presto query and transaction in that case.